### PR TITLE
use noreferrer for window.open with external URLs

### PR DESCRIPTION
**Pull request by diracdeltas**
fix https://github.com/brave/brave-browser/issues/28700
excludes `components/brave_wallet_ui/utils/block-explorer-utils.ts` which is fixed in https://github.com/brave/brave-core/pull/17333/files

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/ipfs_ui/reducers/ipfs_reducer.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ const openURlInNewTab = (target: string, api: string) => {
     return
   }
   const port = api.slice(api.lastIndexOf('/') + 1, api.length)
-  window.open(target.replace('{api-port}', port), '_blank')
+  window.open(target.replace('{api-port}', port), '_blank', 'noreferrer')
```
Comment (diracdeltas) [inline components/ipfs_ui/reducers/ipfs_reducer.ts:24]: cc @lidel @yrliou - i don't think this would break any ipfs functionality but please check

Comment (yrliou) [reply components/ipfs_ui/reducers/ipfs_reducer.ts:24]: cc @cypt4 to double check

**Review by thypon — APPROVED** [review]

**Comment by thypon** [comment]
Checking for missing occurrences

**Review by thypon — CHANGES_REQUESTED** [review]
EDIT all good besides

```
components/brave_new_tab_ui/components/default/braveToday/cards/_articles/cardArticleLarge.tsx
     security-action.assets.semgrep_rules.web.missing-noopener-window-open-native
        window.open should enforce `noopener` to avoid tab-nabbing vulnerabilities.

         42┆ document.open(promotedInfoUrl, '__blank')
```

**Comment by diracdeltas** [comment]
@thypon please see the PR description.

> excludes components/brave_wallet_ui/utils/block-explorer-utils.ts which is fixed in https://github.com/brave/brave-core/pull/17333/files

doing it here will cause a rebase issue

the other ones are opening chrome://, i don't think we need to enforce that so it'd be best if the semgrep rule ignored chrome:// openings. but i can fix it if it's easier.

**Comment by thypon** [comment]
@diracdeltas fixed the ruleset. It's probably better to fix the `document.open` too as mentioned here:

screenshot_2023-02-23_at_11 36 17

**Review by iefremov — APPROVED** [review]
patches LGTM

**Review by emerick — APPROVED** [review]
Rewards changes LGTM

File: components/ipfs_ui/reducers/ipfs_reducer.ts (hunk 1 above)
Comment (cypt4) [reply components/ipfs_ui/reducers/ipfs_reducer.ts:24]: Checked, it is working

**Review by cypt4 — APPROVED** [review]
LGTM

**Review by thypon — APPROVED** [review]

File: components/brave_new_tab_ui/components/default/braveToday/cards/_articles/cardArticleLarge.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ const promotedInfoUrl = 'https://brave.com/brave-today'
 function onClickPromoted (e: React.MouseEvent) {
   const openInNewTab = e.ctrlKey || e.metaKey
   if (openInNewTab) {
-    document.open(promotedInfoUrl, '__blank')
+    document.open(promotedInfoUrl, '__blank', 'noreferrer')
```
Comment (diracdeltas) [inline components/brave_new_tab_ui/components/default/braveToday/cards/_articles/cardArticleLarge.tsx:42]: @petemill is this ok? with 3 arguments this behaves like window.open

Comment (diracdeltas) [reply components/brave_new_tab_ui/components/default/braveToday/cards/_articles/cardArticleLarge.tsx:42]: see https://developer.mozilla.org/en-US/docs/Web/API/Document/open which implies the two-argument window.open form is just like calling `document.open()`, i'm not sure that's what you want

**Review by josheleonard — APPROVED** [review]
LGTM

**Pull request merged**